lib/chainsync/chainsync/test_fixtures/db_session.py

```diff
@@ -81,29 +83,33 @@ def psql_docker() -> Iterator[PostgresConfig]:
         detach=True,
         remove=True,
     )
-
-    # Wait for the container to start
-    time.sleep(3)
+    assert isinstance(container, Container)
 
     # Get version of postgres
-    version_out = container.exec_run("postgres -V")[1]  # type:ignore
+    connected = False
+    version_out = ""
+    while not connected:
```

Comment:
this seems dangerous -- I think we should set a `max_retries` that can be set via an environment variable. @slundqui thoughts?

Reply:
Agreed, this should check a certain amount of times before exiting, with a sleep between checks

Reply:
we have [for _ in range(10):](https://github.com/delvtech/agent0/blob/df58e5ecd5c1b15e6e474df910951e044aea3016/lib/chainsync/chainsync/db/base/interface.py#L99) in interface.py, so i'll just re-use that logic